week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,39 @@
-function setAlarm() {}
+function setAlarm() {
```

Comment:
The basic idea is not bad in this function and your structured your code very well, the presented a separation of concern is also great. However, there is a fundamental concept issue saying that you set up your code each time when you click on the Set button. After each clicks when this listener runs:
```
document.getElementById("set").addEventListener("click", () => {
    setAlarm();
  });
```
it attaches a new set of event listeners to your buttons inside your function without removing the old ones. This means that as many times you click the Set button, as many event listeners you will have on each button, as a result of this, when you click a button, all of the attached event listeners on that will be triggered and executed.

The `setAlarm` function should be responsible to either remove the listeners before adding new ones or prevent adding again if they are already attached.

Have you tried to test your code? It is very easy to break. Try to pay attention to the testing side. A good strategy can be defining different test scenarios that will help you focus on one thing at a time. 
For example:
 1. when I set value and click the set button then time should start
 2. when I set a value smaller than 10, seconds should be padded to 2-character long with a leading zero.
 3. when the timer is up, the counter should stop and alarm clock start

Reply:
Try to use the debugger to inspect and understand better how the code behaves and why. Does it work as expected or not? Alternatively, you can use console log that would also give some clue telling what is happening at a given momentum